src/pykiso/lib/connectors/trc_handler.py

```diff
@@ -0,0 +1,325 @@
+            msg.arbitration_id = int(cols[self.columns["I"]], 16)
+            msg.is_extended_id = len(cols[self.columns["I"]]) > 4
+        # No arbitration id for ST / ER and EC messages
+        else:
+            msg.arbitration_id = cols[self.columns["I"]]
+            msg.is_extended_id = False
+
+        msg.channel = int(cols[bus]) if bus is not None else 1
+        msg.dlc = dlc
+        msg.data = bytearray(
+            [int(cols[i + self.columns["D"]], 16) for i in range(length)]
+        )
+        msg.is_rx = cols[self.columns["d"]] == "Rx"
+        msg.is_fd = type_ in ["FD", "FB", "FE", "BI"]
+        msg.bitrate_switch = type_ in ["FB", " FE"]
+        msg.error_state_indicator = type_ in ["FE", "BI"]
+        msg.is_error_frame = type_ == "ER"
+        msg.is_remote_frame = type_ == "RR"
+
+        return TypedMessage(type_, msg)
+
+
+class TRCWriterCanFD(TRCWriter):
+    """Logs CAN FD data to text file (.trc)"""
+
+    # Type has been added to FORMAT_MESSAGE >= 2.0
+    FORMAT_MESSAGE_V2_1 = "{msgnr:>7} {time:13.3f} {type:>2} {channel:>2} {id:>8} {dir:>2} - {dlc:<4} {data}"
+
+    FORMAT_MESSAGE_V2_0 = (
+        "{msgnr:>7} {time:13.3f} {type:>2} {id:>8} {dir:>2} {dlc:<2} {data}"
+    )
+
+    def _format_message_init(self, msg: TypedMessage, channel: int) -> str:
+        """Pick message format from file version and format initial message
+
+        :return: message informations formated in a string
+        """
+        # Fix error from python can -> message number should start at one not zero
+        self.msgnr = 1
+
+        if self.file_version == TRCFileVersion.V1_0:
+            self._msg_fmt_string = self.FORMAT_MESSAGE_V1_0
+            self._format_message = self._format_message_by_format
+            return self._format_message_by_format(msg, channel)
+        elif self.file_version == TRCFileVersion.V2_0:
+            self._msg_fmt_string = self.FORMAT_MESSAGE_V2_0
+            self._format_message = self._format_message_by_format
+            return self._format_message_by_format(msg, channel)
+        elif self.file_version == TRCFileVersion.V2_1:
+            self._msg_fmt_string = self.FORMAT_MESSAGE_V2_1
+            self._format_message = self._format_message_by_format
+            return self._format_message_by_format(msg, channel)
+        else:
+            raise NotImplementedError("File format is not supported")
+
+    def _format_message_by_format(self, msg: TypedMessage, channel: int):
+        """Format messages
+
+        :return: message informations formated in a string
+        """
+        if msg.type not in ["EC", "ER", "ST"]:
+            if msg.is_extended_id:
+                arb_id = f"{msg.arbitration_id:07X}"
+            else:
+                arb_id = f"{msg.arbitration_id:04X}"
+        else:
+            arb_id = f"{msg.arbitration_id}"
+        data = [f"{byte:02X}" for byte in msg.data]
+
+        """For the time python-can was doing substraction with the first message timestamp
```

Comment:
Please make regular comments out of this, it quite confusing to have a docstring-like text block in the middle of a function IMO :D

Reply:
Done